internal/docker/deployer.go

```diff
@@ -380,8 +380,7 @@ func deployImage(
 			"complement_hs_name":   hsName,
 		},
 	}, &container.HostConfig{
-		CapAdd:          []string{"NET_ADMIN"}, // TODO : this should be some sort of option
-		PublishAllPorts: true,
```

Comment:
> This may fix issues where Complement sporadically decides that there are port conflicts, causing CI flakes.

Where are you running into the CI flakes?

I'm curious because previously you said you didn't see them in vanilla complement runs, https://github.com/matrix-org/complement/pull/761